### DISCUSSION TITLE
Storefront Cart Line Items vs Checkout Line Items

### DISCUSSION THREAD

**Discussion by dbzx10299** [Shopify/storefront-api-feedback#154] [Help] [answered]
Can someone please explain to me the reason we can update the checkout line items instead of only doing this on the cart? I have read the docs on cart api and checkout api, I understand graphql well, but  I am having some trouble understanding why there would be any reason to update line items on a checkout object. 

In other words, what is the relationship with the cart object and the checkout object as far as editing and adding items is concerned?

I would expect to update line items in the cart not in the checkout. When I hear checkout, I think of shopify‘s checkout page that collects the address for a customer. I've yet to see a page where the checkout let‘s you update the items so I would really appreciate if someone could point me in the right direction and clarify that. 

One thing adding to this confusion is shopify‘s checkout page can‘t be customized unless you‘re on a shopify plus plan and on the checkout page there is no way to update line items anyways.

**Comment by johncraigcole** [comment, marked as answer]
Hey @dbzx10299 👋 the checkout object was Shopifys initial API object for facilitating a cart on a headless/custom storefront. The object was to some extent an API wrapper around the actual checkout object being used during Shopify's web checkout.  Overtime as we got feedback and larger merchants started building headless storefronts, Shopify wanted to evolve to meet the scale of our largest merchants while providing new selling models like subscriptions.

The Storefront API cart object was introduced in 2021 to do all that and is the current recommended approach for facilitating commerce on a custom storefront. It's also what we use in our Hydrogen open source framework. 

The Storefront API cart is what you expect. An object to manage what you might want to purchase. Adding, updating, removing items or adding discounts are all possible. While the cart object is linked to the Shopify web checkout that results from a buyer visiting the `checkoutUrl` of a cart object, you can't modify the resulting checkout object via API. You're correct modifications of a checkout is exclusive to Plus merchants. With checkout extensions you can do things like upsell during the checkout flow.

Hope this helps, if you have another questions please let me know.
